WordPress/gutenberg
# Migrate store/actions from editor package to fields package

## What?


This PR continues the work started in [PR #65261](https://github.com/WordPress/gutenberg/pull/65261) by migrating the remaining actions, with the exception of `DuplicateTemplatePart`, which will be addressed in a separate PR.

Changes Introduced

To avoid code duplication and standardize the way actions are written in the codebase, this PR introduces two new functions in the `mutation` folder:
- deletePostWithNotices
- editPostWithNotices

Happy to revisit the naming of these functions! 

These functions handle the updating of entities and manage success and error notices. The logic they encapsulate was previously duplicated across multiple actions, so unifying it into these functions should improve maintainability.​


Currently, there are other entities in this package that update entities without using these two utils. I'm going to refactor them once this PR is merged. 

### Old description

<details>

This PR is a follow-up of https://github.com/WordPress/gutenberg/pull/65261. With this PR, all actions have been migrated except for:

## `deletePostAction`

The blocker for this action is the dispatch of `removeTemplates` action imported by the `@wordpress/editor` package:

https://github.com/WordPress/gutenberg/blob/4356a42b05e569536502f33fd41b69a4e23722f5/packages/editor/src/dataviews/actions/delete-post.tsx#L53-L55

## `duplicateTemplatePart` 

The blocker for this action is `CreateTemplatePartModalContents`:

https://github.com/WordPress/gutenberg/blob/4356a42b05e569536502f33fd41b69a4e23722f5/packages/editor/src/dataviews/actions/duplicate-template-part.tsx#L53-L54


This component is used in the `@wordpress/editor` package by another component.

## `resetPost`

The blocker for this action is the dispatch of `removeTemplates` action imported by the `@wordpress/editor` package:

https://github.com/WordPress/gutenberg/blob/4356a42b05e569536502f33fd41b69a4e23722f5/packages/editor/src/dataviews/actions/reset-post.tsx#L44-L46

<hr />

I'm wondering how we should tackle this. Should we re-implement `removeTemplates` in @wordpress/fields? 🤔 

cc @youknowriad @oandregal 

</details>


## Testing Instructions

Ensure that `Custom Dataviews` is enabled.

1. Visit the Site Editor.
2. Visit the patterns view.
3. Smoke test all the actions available.
4. Visit the templates view.
5. Smoke test all the actions available.
6. Visit the page view.
7. Smoke test all the actions available.


### Testing Instructions for Keyboard


## Screenshots or screencast
